### PR TITLE
TKNECO-30: Skopeo-Copy Task

### PR DESCRIPTION
**Pull request by anandxkumar**
Created a draft template for Skopeo-Copy Task which takes input parameters (Following ALL_CAPS scheme)  as:

- `SOURCE_REGISTRY` - for storing source registry URL
- `DESTINATION_REGISTRY` -  for storing source registry URL
- `SOURCE_AUTH_USERNAME` - for storing source registry username for authentication purpose
- `SOURCE_AUTH_PASSWORD` - for storing source registry password
- `DESTINATION_AUTH_USERNAME` - for storing destination registry username
- `DESTINATION_AUTH_PASSWORD` - for storing destination registry password

Currently the task comprises of 3 steps:

- Authentication of Source directory
- Authentication of Destination directory
- Skopeo-Copy task which basically runs the script for copying the image from one registry to another

Upon testing the task by TaskRun, I was able to copy image from source to destination. 

But it is just a draft implementation, I guess the way to proceed with it ahead will be:

- Instead of writing the steps scripts manually in the `task-sc.yaml` we have separate bash scripts doing the same
- Will we be requiring `workspaces` for this task since skopeo-copy doesn't involve storing image locally? Also how will the authentication take place like in which mode we will get the credentials?
-  Bundling it in Helm Package
- Create Integration tests and e2e test and verify it locally using bats.

Let me know your thoughts and how I should proceed with the PR.

### CODE REVIEW

File: templates/task-sc.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+    name: skopeo-copy
```
Comment (otaviof) [inline templates/task-sc.yaml:5]: Could you please check the settings of your editor? It should have the tab-stop set to two spaces.

File: templates/task-sc.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    labels:
+        app.kubernetes.io/version: 0.0.1
+spec:
+    description: | 
+        Skopeo is a command line tool for working with remote image registries.
```
Comment (otaviof) [inline templates/task-sc.yaml:10]: This line is too long, usually the description is limited to 80 columns to be nicely displayed later on. Please consider [this example](https://github.com/openshift-pipelines/task-git/blob/fbd3c8b15b10b0deda87538a8cbfb53aada6b1d3/templates/task.yaml#L18-L21).

**Review by otaviof — CHANGES_REQUESTED** [review]
Thanks for the PR, overall it looks good with some modifications needed to continue working on it. Thanks in advance!

File: templates/task-sc.yaml (hunk 1 above)
Comment (anandxkumar) [reply templates/task-sc.yaml:5]: Okay will edit it.

File: templates/task-sc.yaml (hunk 2 above)
Comment (anandxkumar) [reply templates/task-sc.yaml:10]: Got it!

**Review by vdemeester — APPROVED** [review]

**Review by otaviof — APPROVED** [review]

**Pull request merged**